### ISSUE TITLE
Android Pie VPN DNS

### ISSUE THREAD

**Issue by CloudEric** [schwabe/ics-openvpn#997]
1. Android Version 9
2. Android Vendor/Custom ROM OnePlus
3. Device 5T
4. Version of the app (version number/play store version/self-built) 0.7.5
 
Hi I just upgraded my phone to Android Pie and OpenVPN for Android seems to have stopped forwarding most packets. I use it to connect to a split tunnel openVPN server that I created to filter DNS. 

VPN works fine on my home PC and it worked on my phone with 8.1 before I upgraded. I tested with OpenVPN Connect and it appears to work on my phone with Android Pie.

I can test with dnsleaktest.com and see that it's not working. 
I can look at my DNS server and see that it gets some connections like googleapis.com and other applications but none of the Chrome browser queries.
There doesn't seem to be any difference in testing my phone at home, mobile, and at work.
I currently have no "Allowed Apps" excluded (default).

I saw [issue 988](https://github.com/schwabe/ics-openvpn/issues/988) but I verified my server mtu is the default 1500 and changing this doesn't seem to have any effect.

**Comment by schwabe** [comment]
you could start by posting a log.

**Comment by CloudEric** [comment]
Thanks for your response, any help would be appreciated.
[openvpnforandroidlog.txt](https://github.com/schwabe/ics-openvpn/files/2810620/openvpnforandroidlog.txt)

**Comment by schwabe** [comment]
```
2019-01-29 20:57:21 Local IPv4: 192.168.255.6/30 IPv6: null MTU: 1500
2019-01-29 20:57:21 DNS Server: 172.17.0.2, Domain: null
2019-01-29 20:57:21 Routes: 172.17.0.0/16, 192.168.255.1/32, 192.168.255.4/30
2019-01-29 20:57:21 Routes excluded: 10.0.0.5/24
2019-01-29 20:57:21 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
2019-01-29 20:57:21 VpnService routes installed: 172.17.0.0/16, 192.168.255.1/32, 192.168.255.4/30
```

This part of the log is what my app tells Android to do. That looks right. What I don't understand is what OpenVPN connect is behaving different in your case. Can you try selecting the OpenVPN3 core under general setting and see if that makes a difference and maybe share a log from the OpenVPN Connect app if possible so I can see what it does differently.

**Comment by CloudEric** [comment]
I think I resolved the issue. I don't know if this is something that can be addressed with your application or even if it's appropriate as an option but I believe it's a major issue with all VPN connections on Android moving forward. At the very least, it should be well documented for anybody expecting VPN functionality. The issue appears to be with a new feature in Chrome called "Async DNS resolver" that ignores your DNS settings and forwards queries to Google. The issue is documented [here](https://www.androidsage.com/2018/01/25/how-to-fix-adhell-2-not-working-chrome/). 

I solved my issue with the following steps: 

- Launch Chrome
- Enter the URL "chrome://flags"
- search for "dns"
- Set "Async DNS resolver" to "Disabled"
- Click on "RELAUNCH NOW" to restart your browser

Thanks for your help with this schwabe and thanks for the excellent application!
